File: core/src/main/java/heast/core/network/packets/c2s/auth/LoginC2SPacket.java
package heast.core.network.packets.c2s.auth;

import heast.core.network.Packet;
import heast.core.network.PacketBuf;
import heast.core.network.listeners.ServerAuthListener;

public final class LoginC2SPacket implements Packet<ServerAuthListener> {

    private final String email;
    private final String password;

    public LoginC2SPacket(String email, String password) {
        this.email = email;
        this.password = password;
    }

    public LoginC2SPacket(PacketBuf buf) {
        this.email = buf.readString();
        this.password = buf.readString();
    }

    @Override
    public void write(PacketBuf buf) {
        buf.writeString(email);
        buf.writeString(password);
    }

    @Override
    public void apply(ServerAuthListener listener) {
        listener.onLogin(this);
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }
}

File: Core/Network/Packets/C2S/LoginC2SPacket.cs
﻿using Core.Network.Listeners;

namespace Core.Network.Packets.C2S;

public class LoginC2SPacket : IPacket
{

	public LoginC2SPacket(string usernameOrEmail, string password)
	{
		UsernameOrEmail = usernameOrEmail;
		Password = password;
	}

	public LoginC2SPacket(PacketBuf buf)
	{
		UsernameOrEmail = buf.ReadString();
		Password = buf.ReadString();
	}

	public string UsernameOrEmail { get; }
	public string Password { get; }

	public void Write(PacketBuf buf)
	{
		buf.WriteString(UsernameOrEmail);
		buf.WriteString(Password);
	}

	public void Apply(IPacketListener listener)
	{
		if (listener is IServerAuthListener authListener)
		{
			authListener.OnLogin(this);
		}
	}
}